### PR TITLE
Also accept an RSA key with crlf

### PR DESCRIPTION
**Pull request by raarts**
I had a private/public key pair containing crlf endings. It was passed in through an environment variable. It wasn't recognized by the RSA_KEY_PATTERN in the CryptKey class. Here's a fix.

### CODE REVIEW

**Comment by m-bymike** [comment]
@raarts Hi, I added a similar pull request in #731 boosting performance of the regular expression and support crlf. We should combine them somehow, because my PR doesn't come with a separate test...

**Comment by raarts** [comment]
Need to discuss with the maintainers? I propose you remove the '\n' from the regex, and we work to get it merged, I'll remove my edits on the regex, and merge the test, and then we'll improve the tests?

**Comment by m-bymike** [comment]
@raarts I suggest you copy the expression from my PR and commit it to yours. I'll simply close my PR...

**Comment by raarts** [comment]
Ok, will do. Thanks.

**Comment by Sephster** [comment]
Thanks both. I think it is probably fine to merge these two PRs together as @m-bymike suggested because they are largely addressing the same issue. I will try and take a look at this later this evening. Thanks for your contribution

File: src/CryptKey.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
 class CryptKey
 {
     const RSA_KEY_PATTERN =
-        '/^(-----BEGIN (RSA )?(PUBLIC|PRIVATE) KEY-----\n)(.|\n)+(-----END (RSA )?(PUBLIC|PRIVATE) KEY-----)$/';
```
Comment (Sephster) [inline src/CryptKey.php]: I'm having a bit of trouble testing this change. When pulling this branch, the crlf file created didn't contain crlf line breaks when I ran `cat -e private.key.crlf`. I converted the file using unix2dos to get crlf line endings and with a bit of tweaking to the tests, found that the original regex pattern still matched. It is worth noting that the new regex also matched. There might be something I am missing here though so apologies if this is the case!

**Comment by Sephster** [comment]
Hi @raarts - Do you have time to look at my comments? Wondering what your thoughts are about the changes I've suggested? Many thanks

File: .gitattributes [hunk 2]
```diff
@@ -10,4 +10,6 @@
 /phpunit.xml.dist export-ignore
 /CHANGELOG.md export-ignore
 /CONTRIBUTING.md export-ignore
-/README.md export-ignore
\ No newline at end of file
+/README.md export-ignore
+
++*.crlf eol=crlf
```
[the remark this answers is not in the document]
Comment (raarts) [reply .gitattributes]: Agreed.

File: src/CryptKey.php (hunk 1 above)
Comment (raarts) [reply src/CryptKey.php]: There's an error in the .gitattributes file, the llne I added starts with a '+'. I have fixed it in a new commit.

File: tests/CryptKeyTest.php [hunk 3]
```diff
@@ -21,6 +21,12 @@ public function testKeyCreation()
 
         $this->assertEquals('file://' . $keyFile, $key->getKeyPath());
         $this->assertEquals('secret', $key->getPassPhrase());
+
+        $keyFile = __DIR__ . '/Stubs/private.key.crlf';
```
[the remark this answers is not in the document]
Comment (raarts) [reply tests/CryptKeyTest.php]: You're right, I misunderstood the tests. Fixed.

**Comment by Sephster** [comment]
Hi @raarts, thanks for making the changes. This is working for me now and I can confirm that your change accepts crlf line endings. 

The only thing left that I have noticed is that we aren't explicitly matching a line ending after the header and I believe, the header should be on its own line as per the [associated rfc](https://tools.ietf.org/html/rfc4716#section-3.2).

At the moment, the changed regular expression rule will match the following:

> -----BEGIN RSA PRIVATE KEY-----A-----END RSA PRIVATE KEY-----

I don't believe the above is a valid key format. Would you be able to adjust the rule slightly to match specific line endings? e.g. <CR>, <LF>, or <CR><LF>.

After that I think this will be good to go. Many thanks for your efforts

**Comment by Sephster** [comment]
Thanks @raarts - apologies for the delay in getting this pushed through and thanks for your patience!

**Pull request merged**